pkg/network/sriov/sriov.go

```diff
@@ -19,4 +19,74 @@
 
 package sriov
 
+import (
+	"encoding/json"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
 const AliasPrefix = "sriov-"
+
+func CreateNetworkPCIAnnotationValue(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (string, error) {
+	networkPciMap, err := mapNetworkNameToPciAddress(networks, interfaces, networkStatusAnnotationValue)
+	if err != nil {
+		return "", err
+	}
+
+	if len(networkPciMap) == 0 {
+		return "", fmt.Errorf("expecting non-emtpy network-pci-map")
+	}
+	networkPciMapBytes, err := json.Marshal(networkPciMap)
+	if err != nil {
+		return "", err
+	}
+
+	return string(networkPciMapBytes), nil
+}
+
+func mapNetworkNameToPciAddress(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (map[string]string, error) {
+	multusInterfaceNameToNetworkStatusMap, err := mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue)
+	if err != nil {
+		return nil, err
+	}
+	networkNameScheme := namescheme.CreateNetworkNameScheme(networks)
+
+	networkPciMap := map[string]string{}
+	for _, sriovIface := range vmispec.FilterSRIOVInterfaces(interfaces) {
+		multusInterfaceName := networkNameScheme[sriovIface.Name]
+		networkStatusEntry := multusInterfaceNameToNetworkStatusMap[multusInterfaceName]
```

Comment:
IIUC `k8s.v1.cni.cncf.io/network-status` creates an entry for each entry in `k8s.v1.cni.cncf.io/networks`
Since we explicitly create the entries of  `k8s.v1.cni.cncf.io/networks` [here](https://github.com/kubevirt/kubevirt/pull/8226/files#diff-d317cef6f369fd934b3c1086561948ea5a267dc3e5beaf38898be013d088a01aL59) using the same nameScheme function - then we are guaranteed that `multusInterfaceNameToNetworkStatusMap[multusInterfaceName]` exists, since 
multusInterfaceName is a sriovIface.Name which is a subset of the SRIOV networks...

Reply:
but we can always have bugs on Multus so I'll address this issue

Reply:
Done. Will now fail if not any SRIOV networks is not showing on the network-status list